### ISSUE TITLE
STAS-F07: Areas Inaccessible via Keyboard

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#923]

**Comment by AlOneill** [comment]
Have removed `<abbr>` markup from public pages, About and Help. There is no satisfactory/foolproof way to make abbr titles accessible to keyboard-only users in free-flowing text.

Have *not* removed from Info for Transcribers/Coords pages. Will continue to think on how best to present such information for volunteers, bearing in mind that  a keyboard-only user could be a volunteer in any role.

**Comment by AlOneill** [comment]
As for the Cookie Notice, I had thought it was *all* out of our direct control, but seeing Doug's recent comment about correcting the link, I had a good look at:
https://github.com/FreeUKGen/MyopicVicar/blob/master/app/views/layouts/application.html.erb

The JavaScript that begins on line 131 gives us control over the **text size** (currently way too big) and the **colours** (which currently do not work well together).

May I suggest that we (i.e. one of the developers!) make the following changes:

* `fontSize: '0.8rem'` ---a little smaller than default text size
* `fontColor: '#444444'` -- default text colour
* `backgroundColor: '#f4f4f4'` -- light grey
* `linkColor: 'a1185a'` -- our "magenta"

It could be that the size of the link text remains large (cannot test myself), but as a *temporary* fix, the notice would be improved.

**Comment by AlOneill** [comment]
One more property to change/remove:
`backgroundOpacity` needs to be 100% (not at all see-through). Probably easier to just remove.

**Comment by AlOneill** [comment]
Now that urgent changes to cookie notice are in place, there remain some issues for Accessibility:

* the checkbox and Continue button are not reachable for a keyboard-only user. This is a source-order issue. -- the cookie notice is *last* in the source-order. I now have a simple solution which I'm 99% sure will work.

* the checkbox needs a label to be Accessible to a screen-reader user. Straightforward to fix. This will also increase the clickable area for toggling the (small) checkbox.

* to ensure best contrast with the background orange, text colour should be `#282828` or darker

**Comment by Captainkirkdawson** [comment]
I await your changes. in mean time will deploy the response to 887.

@AlOneill  Are there other areas than the cookie that fall under this story?

**Comment by AlOneill** [comment]
Not sure how best to describe changes needed -- all in all a bit twiddly. Can you provide a link to the document that needs editing (the Cookie Directive?)

No: the Shaw Report also listed Abbreviations (now removed from public pages) and Tooltips (being dealt with on a page-by-page basis in other stories).

**Comment by Captainkirkdawson** [comment]
You mean this?

https://github.com/FreeUKGen/MyopicVicar/blob/master/app/assets/javascripts/jquery.cookiesDirective.js#L1-L272

**Comment by AlOneill** [comment]
Thanks. Looks like lines 198 to 201 are the lines to be edited.
Would you like me to edit the file, or send you the changes for you to do the editing?

**Comment by Captainkirkdawson** [comment]
Not sure I understand.

If you are able to edit the file and commit the changes to github so that we can pull them into our servers for checkout please so do.

If not you can either send me an edited file or the edits. Makes no real difference to me

**Comment by AlOneill** [comment]
I am not able to to commit -- have only used GitHub before in a simple, one-person situation, and even then was not entirely comfortable with all the processes, being a bear of small brain!!

Will edit file and send to you, Kirk.

**Comment by Captainkirkdawson** [comment]
Completed

**Comment by AlOneill** [comment]
Turns out the cookie notice is not accessible to screen-readers: will create a new story.
